frontend/src/components/settings/components/users-settings/users-settings.tsx

```diff
@@ -52,8 +53,14 @@ export const UsersSettings: React.FC = () => {
     setDeleteModalShown(false);
   };
 
-  const handleSelectChange = (userId: string, role: RoleType): void => {
-    dispatch(usersActions.chageRole({ userId, role }));
+  const handleSelectChange = (
+    userId: string,
+    role: RoleType,
+    fullname: string,
+  ): void => {
+    dispatch(usersActions.chageRole({ userId, role })).then(() => {
+      toast.info(`${fullname} role in the workplace has changed to ${role}`);
+    });
```

Comment:
Promise from dispatch is always resolved, therefore `.then` handler will execute even if server rejected the action. To get the underlying promise that either resolves to **fulfilled** or **rejected** you need to call `.unwrap()` on `dispatch`.

Right now this toast with success message is displayed even if we could not change the user's role because of some error.

```suggestion
    dispatch(usersActions.chageRole({ userId, role }))
      .unwrap()
      .then(() => toast.info(`${fullname} role in the workplace has changed to ${role}`))
      .catch(() => toast.error('An error happened while changing role of the user'));
```